### PR TITLE
[BD-6] Compliance with OEP-18

### PR DESCRIPTION
**Pull request by ericfab179**
Update travis file using new requirements.
Update tox file using new requirements.
Remove old requirement files.
Create requirements folder.
create pip_tools, base, test, travis and tox pip-tools requirements files.
Create Makefile and add upgrade command.
Include requirements files generated using upgrade command.
Update openedx.yml OEPs list.

**Reviewers:**
- [x] @morenol
- [ ] @awais786 
- [ ] @jmbowman

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ericfab179! I've created [OSPR-4436](https://openedx.atlassian.net/browse/OSPR-4436) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by morenol** [comment]
@ericfab179 I think that after a rebase we are ready

File: tests/data/test_do_not_detect_changes.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+# This file is used by the test_do_not_detect_changes, which verifies the detect_changes method doesn't detect changes in rows that do not start with msgid or msgstr.
```
Comment (morenol) [inline tests/data/test_do_not_detect_changes.txt:1]: :+1:

File: tox.ini [hunk 2]
```diff
@@ -5,7 +5,7 @@ envlist = py35-django22,py38-django{22,30}, quality
 deps =
     django22: Django>=2.2,<2.3
     django30: Django>=3.0,<3.1
-    -rtest_requirements.txt
+    -r requirements/test.txt
```
Comment (morenol) [inline tox.ini:8]: I think that you should remove that whitespace added in here

**Comment by ericfab179** [comment]
@awais786 We think that this PR is ready to be merged.

**Comment by edx-webhook** [comment]
@ericfab179 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/i18n-tools&entry.867055334=https://github.com/edx/i18n-tools/pull/85&entry.1484655318=https://github.com/ericfab179&entry.752974735=2020-04-30+00:53&entry.1917517419=2020-05-11+14:53&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**